packages/native-core/src/state.rs

```diff
@@ -92,6 +138,7 @@ pub trait NodeDepState {
     ) -> bool;
 }
 
+/// Do not implement this trait. It is only meant to be derived and used through [RealDom].
```

Comment:
also consider `#[doc(hidden)]` if this is an internal implementation detail

Reply:
(it's unclear to me; perhaps it's meant to be used but not implemented? if so, disregard comment)